contracts/test/vanity_url.ts

```diff
@@ -0,0 +1,311 @@
+  wrapperExpiry: maxWrapperExpiry,
+  fuses: config.fuses,
+  registrarController: config.registrarController,
+  nameWrapper: config.nameWrapper,
+  baseRegistrar: config.registrar,
+  resolver: config.resolver,
+  reverseRecord: config.reverseRecord,
+  duration: config.duration
+}
+const dotName = 'test.1.country'
+const urlUpdatePrice = ethers.utils.parseEther("1");
+
+const increaseTime = async (sec: number): Promise<void> => {
+  await network.provider.send("evm_increaseTime", [sec]);
+  await network.provider.send("evm_mine");
+};
+
+const getTimestamp = async (): Promise<BigNumber> => {
+  const blockNumber = await ethers.provider.getBlockNumber();
+  const block = await ethers.provider.getBlock(blockNumber);
+  return BigNumber.from(block.timestamp);
+};
+
+describe('VanityURL', () => {
+  let accounts: SignerWithAddress;
+  let deployer: SignerWithAddress;
+  let alice: SignerWithAddress;
+  let bob: SignerWithAddress;
+  let revenueAccount: SignerWithAddress;
+
+  let mockDC: MockDC;
+  let vanityURL: VanityURL;
+
+  beforeEach(async () => {
+    accounts = await ethers.getSigners();
+    [deployer, alice, bob, revenueAccount] = accounts;
+
+    // Deploy MockDC contract
+    const MockDC = await ethers.getContractFactory('MockDC');
+    mockDC = (await MockDC.deploy()) as MockDC;
+
+    // Deploy VanityURL contract
+    const VanityURL = await ethers.getContractFactory("VanityURL");
+    vanityURL = (await VanityURL.deploy(mockDC.address, urlUpdatePrice, revenueAccount.address)) as VanityURL;
+  });
+
+  describe("setRevenueAccount", () => {
+    it("Should be able set the revenue account", async () => {
+      expect(await vanityURL.revenueAccount()).to.equal(revenueAccount.address);
+      
+      await vanityURL.setRevenueAccount(alice.address);
+
+      expect(await vanityURL.revenueAccount()).to.equal(alice.address);
+    });
+
+    it("Should revert if the caller is not owner", async () => {
+      await expect(vanityURL.connect(alice).setRevenueAccount(alice.address)).to.be.reverted;
+    });
+  });
+
+  describe("setNewURL", () => {
+    const tokenId = ethers.utils.keccak256(ethers.utils.toUtf8Bytes(dotName));
+    const aliasName = "aliasName";
+    const url = "url";
+    const price = ethers.utils.parseEther("2");
+
+    beforeEach(async () => {
+      await mockDC.connect(alice).register(dotName);
+    });
+
+    it("Should be able to set a new URL", async () => {
+      expect(await vanityURL.vanityURLs(tokenId, aliasName)).to.equal("");
+      expect(await vanityURL.vanityURLUpdatedAt(tokenId, aliasName)).to.equal(0);
+
+      // set a new URL
+      await vanityURL.connect(alice).setNewURL(dotName, aliasName, url, price, { value: urlUpdatePrice });
+
+      expect(await vanityURL.vanityURLs(tokenId, aliasName)).to.equal(url);
+      expect(await vanityURL.vanityURLPrices(tokenId, aliasName)).to.equal(price);
+      expect(await vanityURL.vanityURLUpdatedAt(tokenId, aliasName)).to.equal(await getTimestamp());
```

Comment:
I think this may have a small chance to fail if environment is set to mine at regular intervals

Reply:
Would you elaborate on it?